### ISSUE TITLE
issue with dice of AA

### ISSUE THREAD

**Issue by Syssuc** [fengweie/SE_ASA#5]
Dear Wei Feng,

Thanks for your Work, I am currently using your codebase trying to reproduce your model performence. I use senery_CT2MR.yml and ct2mr_model_pretrained.pth to train the model. but the dice of AA even lower than pretrained_model ct2mr_model_pretrained.pth. the dice of AA is a small value in most batches. Am i use the wrong pretrained_model or wrong config?
#################################################################
The test of pretained model:
\SE_ASA-main\SE_ASA-main\scripts>python test.py --target_modality MR --pretrained_model_pth m
odel_30000.pth
target_modality is MR,method is senery
Evaluating model model_30000.pth
pred spend time is 9 seconds
metric spend time is 7 seconds
(256, 256, 130) (5, 256, 256, 130)
pred spend time is 6 seconds
metric spend time is 7 seconds
(256, 256, 130) (5, 256, 256, 130)
pred spend time is 5 seconds
metric spend time is 7 seconds
(256, 256, 130) (5, 256, 256, 130)
pred spend time is 5 seconds
metric spend time is 7 seconds
(256, 256, 130) (5, 256, 256, 130)
(520, 256, 256) (520, 5, 256, 256)
dice arr is (4, 4)
Dice:
AA :40.2(7.3)
LAC:52.6(18.8)
LVC:68.8(5.7)
Myo:47.1(9.4)
Mean:52.2
ASSD:
AA :9.8(2.2)
LAC:27.6(17.1)
LVC:9.5(4.4)
Myo:6.0(2.8)
Mean:13.2
Ece value: tensor([0.0363])
#################################################################
The output during training:
root@autodl-container-e20e41a3dd-9485db6d:~/autodl-fs/scripts# CUDA_VISIBLE_DEVICES=0 python train.py --cfg ./config
s/senery_CT2MR.yml
Called with args
Model loaded from:../pretrain_model/model_30000.pth
dataloader finish
use method: senery
initialize network with normal
initialize network with normal
finish model setup
iter=200 loss_seg_src_aux=0.042 loss_seg_src_main=0.042 loss_dice_aux=0.664 loss_dice_main=0.669 loss_adv_trg_aux=1.235 loss_adv_trg_main=1.030 loss_d_aux=0.226 loss_d_main=0.280 loss_cent_correct=0.018 loss_cent_incorrect=-0.005 loss_infoent=-0.018 loss_class=0.044
iter=400 loss_seg_src_aux=0.033 loss_seg_src_main=0.026 loss_dice_aux=0.612 loss_dice_main=0.609 loss_adv_trg_aux=3.173 loss_adv_trg_main=2.425 loss_d_aux=0.042 loss_d_main=0.060 loss_cent_correct=0.007 loss_cent_incorrect=-0.006 loss_infoent=-0.018 loss_class=0.031
  1%|▋                                                                       | 500/50001 [11:55<19:15:14,  1.40s/it]assd error for class 4: The first supplied array does not contain any binary object.
Dice:
AA :2.1(1.9)
LAC:37.6(18.2)
LVC:74.5(3.8)
Myo:53.8(8.3)
Mean:42.0
ASSD:
AA :18.3(11.9)
LAC:7.6(2.4)
LVC:7.0(3.4)
Myo:4.4(1.9)
Mean:9.3
taking snapshot ...
exp = ../scripts/experiments/snapshots/
iter=600 loss_seg_src_aux=0.049 loss_seg_src_main=0.049 loss_dice_aux=0.419 loss_dice_main=0.418 loss_adv_trg_aux=2.182 loss_adv_trg_main=2.529 loss_d_aux=0.082 loss_d_main=0.055 loss_cent_correct=0.009 loss_cent_incorrect=-0.004 loss_infoent=-0.014 loss_class=0.014
iter=800 loss_seg_src_aux=0.010 loss_seg_src_main=0.009 loss_dice_aux=0.758 loss_dice_main=0.757 loss_adv_trg_aux=4.033 loss_adv_trg_main=2.274 loss_d_aux=0.014 loss_d_main=0.076 loss_cent_correct=0.016 loss_cent_incorrect=-0.011 loss_infoent=-0.035 loss_class=0.018
iter=1000 loss_seg_src_aux=0.063 loss_seg_src_main=0.054 loss_dice_aux=0.454 loss_dice_main=0.447 loss_adv_trg_aux=4.266 loss_adv_trg_main=2.571 loss_d_aux=0.024 loss_d_main=0.089 loss_cent_correct=0.013 loss_cent_incorrect=-0.014 loss_infoent=-0.037 loss_class=0.027
  2%|█▍                                                                     | 1000/50001 [24:36<18:54:20,  1.39s/it]assd error for class 4: The first supplied array does not contain any binary object.
Dice:
AA :1.7(1.7)
LAC:41.4(21.7)
LVC:74.0(3.8)
Myo:52.6(7.5)
Mean:42.4
ASSD:
AA :22.6(21.0)
LAC:8.0(2.3)
LVC:5.6(2.9)
Myo:3.8(1.4)
Mean:10.0
taking snapshot ...
exp = ../scripts/experiments/snapshots/
iter=1200 loss_seg_src_aux=0.031 loss_seg_src_main=0.034 loss_dice_aux=0.613 loss_dice_main=0.614 loss_adv_trg_aux=2.143 loss_adv_trg_main=1.970 loss_d_aux=0.079 loss_d_main=0.088 loss_cent_correct=0.005 loss_cent_incorrect=-0.005 loss_infoent=-0.009 loss_class=0.017
iter=1400 loss_seg_src_aux=0.032 loss_seg_src_main=0.032 loss_dice_aux=0.706 loss_dice_main=0.704 loss_adv_trg_aux=2.571 loss_adv_trg_main=1.816 loss_d_aux=0.066 loss_d_main=0.107 loss_cent_correct=0.003 loss_cent_incorrect=-0.002 loss_infoent=-0.012 loss_class=0.023
  3%|██▏                                                                    | 1500/50001 [37:05<19:24:37,  1.44s/it]Dice:
AA :20.8(10.9)
LAC:58.7(10.4)
LVC:74.2(6.8)
Myo:57.0(7.0)
Mean:52.7
ASSD:
AA :12.9(7.1)
LAC:12.0(5.5)
LVC:9.8(5.6)
Myo:5.7(3.1)
Mean:10.1
taking snapshot ...
exp = ../scripts/experiments/snapshots/
iter=1600 loss_seg_src_aux=0.037 loss_seg_src_main=0.033 loss_dice_aux=0.664 loss_dice_main=0.662 loss_adv_trg_aux=1.425 loss_adv_trg_main=0.465 loss_d_aux=0.163 loss_d_main=0.547 loss_cent_correct=0.006 loss_cent_incorrect=-0.003 loss_infoent=-0.019 loss_class=0.017
iter=1800 loss_seg_src_aux=0.044 loss_seg_src_main=0.035 loss_dice_aux=0.507 loss_dice_main=0.503 loss_adv_trg_aux=2.702 loss_adv_trg_main=0.750 loss_d_aux=0.071 loss_d_main=0.352 loss_cent_correct=0.008 loss_cent_incorrect=-0.004 loss_infoent=-0.019 loss_class=0.029
iter=2000 loss_seg_src_aux=0.044 loss_seg_src_main=0.035 loss_dice_aux=0.517 loss_dice_main=0.513 loss_adv_trg_aux=1.957 loss_adv_trg_main=1.399 loss_d_aux=0.095 loss_d_main=0.176 loss_cent_correct=0.003 loss_cent_incorrect=-0.002 loss_infoent=-0.016 loss_class=0.009
  4%|██▊                                                                    | 2000/50001 [49:44<18:22:16,  1.38s/it]Dice:
AA :13.1(16.7)
LAC:59.8(8.4)
LVC:72.6(4.3)
Myo:56.3(8.3)
Mean:50.5
ASSD:
AA :19.9(14.1)
LAC:13.1(5.3)
LVC:8.1(4.5)
Myo:4.6(2.6)
Mean:11.4
iter=2200 loss_seg_src_aux=0.000 loss_seg_src_main=0.001 loss_dice_aux=0.800 loss_dice_main=0.800 loss_adv_trg_aux=5.328 loss_adv_trg_main=2.680 loss_d_aux=0.004 loss_d_main=0.043 loss_cent_correct=0.005 loss_cent_incorrect=-0.004 loss_infoent=-0.013 loss_class=0.010
iter=2400 loss_seg_src_aux=0.032 loss_seg_src_main=0.029 loss_dice_aux=0.513 loss_dice_main=0.512 loss_adv_trg_aux=3.827 loss_adv_trg_main=2.089 loss_d_aux=0.018 loss_d_main=0.078 loss_cent_correct=0.004 loss_cent_incorrect=-0.002 loss_infoent=-0.010 loss_class=0.008
  5%|███▍                                                                 | 2500/50001 [1:02:12<18:09:27,  1.38s/it]assd error for class 4: The first supplied array does not contain any binary object.
Dice:
AA :5.1(6.3)
LAC:64.6(9.5)
LVC:74.9(2.9)
Myo:57.6(7.4)
Mean:50.5
ASSD:
AA :11.3(7.1)
LAC:5.8(1.9)
LVC:4.1(0.5)
Myo:3.6(1.5)
Mean:6.2
iter=2600 loss_seg_src_aux=0.062 loss_seg_src_main=0.059 loss_dice_aux=0.449 loss_dice_main=0.449 loss_adv_trg_aux=6.300 loss_adv_trg_main=2.463 loss_d_aux=0.002 loss_d_main=0.051 loss_cent_correct=0.007 loss_cent_incorrect=-0.012 loss_infoent=-0.034 loss_class=0.012
iter=2800 loss_seg_src_aux=0.060 loss_seg_src_main=0.052 loss_dice_aux=0.352 loss_dice_main=0.343 loss_adv_trg_aux=5.405 loss_adv_trg_main=4.784 loss_d_aux=0.004 loss_d_main=0.023 loss_cent_correct=0.009 loss_cent_incorrect=-0.006 loss_infoent=-0.023 loss_class=0.008
iter=3000 loss_seg_src_aux=0.033 loss_seg_src_main=0.028 loss_dice_aux=0.708 loss_dice_main=0.708 loss_adv_trg_aux=4.111 loss_adv_trg_main=1.279 loss_d_aux=0.013 loss_d_main=0.190 loss_cent_correct=0.004 loss_cent_incorrect=-0.007 loss_infoent=-0.019 loss_class=0.011
  6%|████▏                                                                | 3000/50001 [1:14:40<18:02:26,  1.38s/it]assd error for class 4: The first supplied array does not contain any binary object.
Dice:
AA :10.9(14.1)
LAC:64.8(9.4)
LVC:77.6(4.5)
Myo:59.4(8.3)
Mean:53.2
ASSD:
AA :10.0(6.5)
LAC:6.3(2.5)
LVC:6.3(3.0)
Myo:4.1(2.1)
Mean:6.7
taking snapshot ...
exp = ../scripts/experiments/snapshots/
iter=3200 loss_seg_src_aux=0.019 loss_seg_src_main=0.014 loss_dice_aux=0.678 loss_dice_main=0.674 loss_adv_trg_aux=4.586 loss_adv_trg_main=1.686 loss_d_aux=0.009 loss_d_main=0.161 loss_cent_correct=0.008 loss_cent_incorrect=-0.006 loss_infoent=-0.034 loss_class=0.010
iter=3400 loss_seg_src_aux=0.053 loss_seg_src_main=0.057 loss_dice_aux=0.416 loss_dice_main=0.416 loss_adv_trg_aux=4.621 loss_adv_trg_main=2.016 loss_d_aux=0.007 loss_d_main=0.087 loss_cent_correct=0.005 loss_cent_incorrect=-0.004 loss_infoent=-0.016 loss_class=0.009
  7%|████▊                                                                | 3500/50001 [1:27:08<18:18:10,  1.42s/it]Dice:
AA :7.4(11.8)
LAC:61.6(9.6)
LVC:77.9(5.1)
Myo:56.9(8.8)
Mean:50.9
ASSD:
AA :59.3(56.1)
LAC:7.0(2.2)
LVC:6.7(3.0)
Myo:4.8(2.6)
Mean:19.5
iter=3600 loss_seg_src_aux=0.037 loss_seg_src_main=0.044 loss_dice_aux=0.562 loss_dice_main=0.563 loss_adv_trg_aux=5.147 loss_adv_trg_main=1.751 loss_d_aux=0.005 loss_d_main=0.144 loss_cent_correct=0.006 loss_cent_incorrect=-0.017 loss_infoent=-0.030 loss_class=0.013
iter=3800 loss_seg_src_aux=0.050 loss_seg_src_main=0.044 loss_dice_aux=0.291 loss_dice_main=0.292 loss_adv_trg_aux=6.432 loss_adv_trg_main=3.280 loss_d_aux=0.001 loss_d_main=0.025 loss_cent_correct=0.005 loss_cent_incorrect=-0.005 loss_infoent=-0.020 loss_class=0.004
iter=4000 loss_seg_src_aux=0.002 loss_seg_src_main=0.001 loss_dice_aux=0.800 loss_dice_main=0.800 loss_adv_trg_aux=1.950 loss_adv_trg_main=2.036 loss_d_aux=0.157 loss_d_main=0.108 loss_cent_correct=0.009 loss_cent_incorrect=-0.009 loss_infoent=-0.045 loss_class=0.006
  8%|█████▌                                                               | 4000/50001 [1:39:37<17:47:20,  1.39s/it]assd error for class 4: The first supplied array does not contain any binary object.
Dice:
AA :6.3(10.0)
LAC:63.6(8.2)
LVC:77.1(4.7)
Myo:55.6(7.8)
Mean:50.7
ASSD:
AA :19.2(18.4)
LAC:5.8(2.1)
LVC:3.8(0.5)
Myo:3.5(1.3)
Mean:8.1
iter=4200 loss_seg_src_aux=0.025 loss_seg_src_main=0.023 loss_dice_aux=0.705 loss_dice_main=0.704 loss_adv_trg_aux=3.249 loss_adv_trg_main=0.400 loss_d_aux=0.030 loss_d_main=0.597 loss_cent_correct=0.009 loss_cent_incorrect=-0.007 loss_infoent=-0.034 loss_class=0.007
iter=4400 loss_seg_src_aux=0.044 loss_seg_src_main=0.035 loss_dice_aux=0.427 loss_dice_main=0.422 loss_adv_trg_aux=5.461 loss_adv_trg_main=5.006 loss_d_aux=0.004 loss_d_main=0.006 loss_cent_correct=0.010 loss_cent_incorrect=-0.008 loss_infoent=-0.039 loss_class=0.006
  9%|██████▏                                                              | 4500/50001 [1:52:02<17:40:03,  1.40s/it]assd error for class 4: The first supplied array does not contain any binary object.
assd error for class 4: The first supplied array does not contain any binary object.
Dice:
AA :7.1(9.2)
LAC:63.4(8.9)
LVC:77.1(5.4)
Myo:57.4(8.1)
Mean:51.3
ASSD:
AA :6.4(5.7)
LAC:6.0(2.0)
LVC:3.8(0.7)
Myo:3.5(1.4)
Mean:4.9
iter=4600 loss_seg_src_aux=0.032 loss_seg_src_main=0.025 loss_dice_aux=0.573 loss_dice_main=0.570 loss_adv_trg_aux=1.993 loss_adv_trg_main=2.111 loss_d_aux=0.108 loss_d_main=0.087 loss_cent_correct=0.006 loss_cent_incorrect=-0.002 loss_infoent=-0.018 loss_class=0.014
iter=4800 loss_seg_src_aux=0.057 loss_seg_src_main=0.048 loss_dice_aux=0.420 loss_dice_main=0.418 loss_adv_trg_aux=3.602 loss_adv_trg_main=0.767 loss_d_aux=0.022 loss_d_main=0.352 loss_cent_correct=0.002 loss_cent_incorrect=-0.004 loss_infoent=-0.007 loss_class=0.008
iter=5000 loss_seg_src_aux=0.037 loss_seg_src_main=0.032 loss_dice_aux=0.510 loss_dice_main=0.510 loss_adv_trg_aux=6.291 loss_adv_trg_main=3.770 loss_d_aux=0.002 loss_d_main=0.016 loss_cent_correct=0.004 loss_cent_incorrect=-0.002 loss_infoent=-0.017 loss_class=0.005
 10%|██████▉                                                              | 5000/50001 [2:04:31<17:31:45,  1.40s/it]assd error for class 4: The first supplied array does not contain any binary object.
Dice:
AA :16.5(18.3)
LAC:64.0(9.0)
LVC:79.7(5.6)
Myo:58.7(8.7)
Mean:54.7
ASSD:
AA :8.5(5.6)
LAC:6.1(2.0)
LVC:4.6(1.7)
Myo:4.0(2.1)
Mean:5.8
taking snapshot ...
exp = ../scripts/experiments/snapshots/
iter=5200 loss_seg_src_aux=0.040 loss_seg_src_main=0.035 loss_dice_aux=0.564 loss_dice_main=0.564 loss_adv_trg_aux=4.912 loss_adv_trg_main=0.415 loss_d_aux=0.008 loss_d_main=0.610 loss_cent_correct=0.006 loss_cent_incorrect=-0.007 loss_infoent=-0.034 loss_class=0.006
iter=5400 loss_seg_src_aux=0.057 loss_seg_src_main=0.039 loss_dice_aux=0.707 loss_dice_main=0.706 loss_adv_trg_aux=4.737 loss_adv_trg_main=2.282 loss_d_aux=0.006 loss_d_main=0.064 loss_cent_correct=0.005 loss_cent_incorrect=-0.003 loss_infoent=-0.018 loss_class=0.012
 11%|███████▌                                                             | 5500/50001 [2:17:00<17:15:33,  1.40s/it]Dice:
AA :8.1(12.5)
LAC:61.4(8.3)
LVC:76.6(7.3)
Myo:54.4(8.4)
Mean:50.1
ASSD:
AA :18.4(10.9)
LAC:8.5(2.4)
LVC:6.9(3.7)
Myo:5.8(3.0)
Mean:9.9
